.github/workflows/foundry.yml

```diff
@@ -0,0 +1,57 @@
+name: foundry
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  # Install foundry and use it to lint and test parts of the codebase that use Solidity.
+  foundry:
+    runs-on: ubuntu-latest
+    steps:
+      # This is a workaround from: https://github.com/actions/checkout/issues/590#issuecomment-970586842
+      - name: checkout dummy commit (submodule bug workaround)
+        run: "git checkout -f $(git -c user.name=x -c user.email=x@x commit-tree $(git hash-object -t tree /dev/null) < /dev/null) || :"
+
+      - name: checkout code
+        uses: actions/checkout@v3
+        with:
+          submodules: recursive
+
+      - name: install rust
+        uses: ./.github/actions/rustup
+
+      - name: Install Foundry
+        uses: risc0/foundry-toolchain@2fe7e70b520f62368a0e3c464f997df07ede420f
+
+      - name: check solidity code formatting in bonsai/ethereum
+        run: forge fmt --check
```

Comment:
could we make this one line? `forget fmt --check bonsai/ethereum`. same with L39?

Reply:
It seems this actually yield different results. It seems in general `forge` expects the project root to be the current directory.